fe2-android/app/src/mock/java/com/github/dedis/student20_pop/Injection.java

```diff
@@ -38,13 +45,50 @@
 
 public class Injection {
 
-  private Injection() {}
+  private static final String KEYSET_NAME = "POP_KEYSET";
+
+  private static final String SHARED_PREF_FILE_NAME = "POP_KEYSET_SP";
+
+  private static final String MASTER_KEY_URI = "android-keystore://POP_MASTER_KEY";
+
+  private static AndroidKeysetManager KEYSET_MANAGER;
+
+  private Injection() {
+  }
 
 
   @SuppressWarnings("unused")
   public static AndroidKeysetManager provideAndroidKeysetManager(Context applicationContext)
       throws GeneralSecurityException, IOException {
-    return null;
+    if (KEYSET_MANAGER == null) {
+      SharedPreferences.Editor editor =
+          applicationContext
+              .getSharedPreferences(SHARED_PREF_FILE_NAME, Context.MODE_PRIVATE)
+              .edit();
+      editor.apply();
+
+      Ed25519PrivateKeyManager.registerPair(true);
+      PublicKeySignWrapper.register();
+
+      // TODO: move to background thread
+      AndroidKeysetManager keysetManager =
+          new AndroidKeysetManager.Builder()
+              .withSharedPref(applicationContext, KEYSET_NAME, SHARED_PREF_FILE_NAME)
+              .withKeyTemplate(Ed25519PrivateKeyManager.rawEd25519Template())
+              .withMasterKeyUri(MASTER_KEY_URI)
+              .build();
+
+      KeysetHandle publicKeysetHandle = keysetManager.getKeysetHandle().getPublicKeysetHandle();
+
+      try {
+        String publicKey = Keys.getEncodedKey(publicKeysetHandle);
+      } catch (IOException e) {
+        Log.e("INJECTION", "failed to retrieve public key", e);
+      }
+
+      KEYSET_MANAGER = keysetManager;
+    }
```

Comment:
does any of this exist in the mocked context? 
should we implement a mock ? is it even feasible ?